iphone/Classes/TiMediaVideoPlayerProxy.m

```diff
@@ -890,40 +907,77 @@ - (void)handleLoadStateChangeNotification:(NSNotification *)note
 
 - (void)handleNowPlayingNotification:(NSNotification *)note
 {
+  _playbackState = TiVideoPlayerPlaybackStatePlaying;
+
   if ([self _hasListeners:@"playing"]) {
     NSDictionary *event = [NSDictionary dictionaryWithObject:[self url] forKey:@"url"];
     [self fireEvent:@"playing" withObject:event];
   }
 }
 
-- (void)handlePlaybackStateChangeNotification:(NSNotification *)note
+- (void)handleNaturalSizeAvailableNotification:(NSNotification *)note
 {
-  if ([self _hasListeners:@"playbackstate"]) {
-    NSDictionary *event = [NSDictionary dictionaryWithObject:[self playbackState] forKey:@"playbackState"];
-    [self fireEvent:@"playbackstate" withObject:event];
+  if ([self _hasListeners:@"naturalsizeavailable"]) {
+    [self fireEvent:@"naturalsizeavailable"
+         withObject:@{
+           @"naturalSize" : @{
+             @"width" : NUMFLOAT(movie.videoBounds.size.width),
+             @"height" : NUMFLOAT(movie.videoBounds.size.height)
+           }
+         }];
   }
+}
+
+// iOS < 10
+- (void)handlePlaybackStateChangeNotification:(NSNotification *)note
+{
+  TiVideoPlayerPlaybackState oldState = _playbackState;
 
   switch ([[movie player] status]) {
   case AVPlayerStatusUnknown:
   case AVPlayerStatusFailed:
     playing = NO;
+    _playbackState = TiVideoPlayerPlaybackStateInterrupted;
     break;
   case AVPlayerStatusReadyToPlay:
     playing = ([[movie player] rate] == 1.0);
+    if (playing) {
+      _playbackState = TiVideoPlayerPlaybackStatePlaying;
+    } else if (movie.player.currentItem.duration.value == movie.player.currentItem.currentTime.value || !movie.player.currentItem.canStepBackward) {
+      _playbackState = TiVideoPlayerPlaybackStateStopped;
+    } else {
+      _playbackState = TiVideoPlayerPlaybackStatePaused;
+    }
     break;
   }
+
+  if ([self _hasListeners:@"playbackstate"] && oldState != _playbackState) {
+    NSDictionary *event = [NSDictionary dictionaryWithObject:[self playbackState] forKey:@"playbackState"];
+    [self fireEvent:@"playbackstate" withObject:event];
+  }
 }
 
-- (void)handleNaturalSizeAvailableNotification:(NSNotification *)note
+// iOS 10+
+- (void)handleTimeControlStatusNotification:(NSNotification *)note
 {
-  if ([self _hasListeners:@"naturalsizeavailable"]) {
-    [self fireEvent:@"naturalsizeavailable"
-         withObject:@{
-           @"naturalSize" : @{
-             @"width" : NUMFLOAT(movie.videoBounds.size.width),
-             @"height" : NUMFLOAT(movie.videoBounds.size.height)
-           }
-         }];
+  TiVideoPlayerPlaybackState oldState = _playbackState;
+  playing = movie.player.timeControlStatus == AVPlayerTimeControlStatusPlaying;
+
+  if (movie.player.timeControlStatus == AVPlayerTimeControlStatusPlaying) {
+    _playbackState = TiVideoPlayerPlaybackStatePlaying;
+  } else if (movie.player.timeControlStatus == AVPlayerTimeControlStatusPaused) {
+    if (movie.player.currentItem.duration.value == movie.player.currentItem.currentTime.value) {
```

Comment:
should this line not be like -
`  if (movie.player.currentItem.duration.value == movie.player.currentItem.currentTime.value || !movie.player.currentItem.canStepBackward) {
` as you have made in 'handlePlaybackStateChangeNotification: '.

Reply:
It is not necessary because it only handles the already running state that cannot interfere with other states - at least not in my tests.